pkg/git/git.go

```diff
@@ -121,6 +121,9 @@ func OpenRepository(ctx context.Context, name, namespace string, spec *configapi
 		return nil, fmt.Errorf("error cloning git repository %q, cannot create remote: %v", spec.Repo, err)
 	}
 
+	// NOTE: the spec.git.branch field in the Repository CRD (OpenAPI schema) is defined with
+	//		 MinLength=1 validation and its default value is set to "main". This means that
+	// 		 it should never be empty at this point. The following code is left here as a last resort failsafe.
 	branch := MainBranch
 	if spec.Branch != "" {
```

Comment:
Is this check still needed if branch is never empty?

Reply:
In theory no, but in practice it may be considered defensive programing :) that prevents potential errors in the future when we change the CRD.  see the comment above.
If the community consensus is to remove this and completely rely on CR validation, I have no objections.

Reply:
Nah leave it as is